private-cloud/ecs-on-aws/README.md

```diff
@@ -0,0 +1,198 @@
+
+* Set up the following definition environment variables:
+
+    | Variable | Description | Status |
+    |----------|-------------|--------|
+    | `SSH_PUBLIC_KEY_FILE` | File path to the SSH public key that will be uploaded to the cloud provider (using the `name_prefix` variable as the key label). E.g. `/Users/example/.ssh/demo_ops.pub` | Mandatory |
+    | `CDP_LICENSE_FILE` | File path to a CDP Private Cloud Base license. E.g. `/Users/example/Documents/example_cloudera_license.txt` | Mandatory |
+    | `IPA_USER` | Set this to `admin`. The adminstrator user for FreeIPA.  | Mandatory |
+    | `IPA_PASSWORD` | The adminstrator and directory password for FreeIPA | Mandatory |
+    | `AWS_PROFILE` | The profile label for your AWS credentials. Otherwise, use the associated `AWS_*` parameters. Used also for remote storage of Terraform state in AWS. | Mandatory |
+
+> **_NOTE:_** For OSX, set the following: `export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES` to allow the WinRM modules to function.
+
+
+#### Configuration file variables
+
+Edit the `config.yml` user-facing configuration file to match your particular deployment.
+
+*NOTE:* `name_prefix` should be 4-8 characters and is the "primary key" for the deployment. `owner_prefix` is used in circumstances to differentiate resources such as the SSH key label in the cloud provider and the subdomain(s) for the private DNS service.
+
+```yaml
+name_prefix:       "labaw"                          # CHANGE THIS
+owner_prefix:      "pvc-ecs"                        
+owner_email:       "example@cloudera.com"            
+infra_region:      "eu-west-1"                      
+infra_type:        "aws"                            # "aws", "static"
+domain:            "{{ owner_prefix }}.cldr.example"   # The private, adhoc subdomain (name_prefix.owner_prefix.cldr.demo)
+realm:             "CLDR.EXAMPLE"                      # The Kerberos realm
+common_password:   "Example776"                   
+admin_password:    "Example776"                   
+deployment_tags:
+  owner: "{{ owner_prefix }}"
+  email: "{{ owner_email }}"
+  project: "PvC ECS Lab - {{ owner_prefix }}-{{ name_prefix }}"
+  enddate: "{{ ('%m%d%Y' | strftime((ansible_date_time.epoch | int) + (90 * 86400))) }}"
+  deployment: "{{ name_prefix }}"
+  deploy-tool: cloudera-deploy
+```
+
+## Execution
+
+### Pre-setup Playbook
+
+This definition-specific playbook includes tasks such as:
+* Instructure provisioning
+* FreeIPA DNS and KRB services provisioning
+
+Run the following command 
+
+```bash
+ansible-navigator run pre_setup.yml \
+-e @config.yml \
+-e @definition.yml
+```
+
+Once the pre-setup playbook completes confirm that:
+
+* You can connect to each node via the inventory - see Confirm SSH Connectivity. Note that a A `validate_dns_lookups.yml` Playbook exists to check connectivity.
+* Connect to FreeIPA UI and login with the `IPA_USER` and `IPA_PASSWORD` credentials in the configuration file. See Cluster Access for more details.
```

Comment:
How does one do this? My FreeIPA node just has port 22 open, to the world, I might add...